src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12354,9 +12390,18 @@ private ParameterSyntax ParseLambdaParameter()
             }
 
             SyntaxToken paramName = this.ParseIdentifierToken();
-            var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
-            exclamation = exclamation is null ? null : CheckFeatureAvailability(exclamation, MessageID.IDS_ParameterNullChecking);
-            var parameter = _syntaxFactory.Parameter(default(SyntaxList<AttributeListSyntax>), modifiers.ToList(), paramType, paramName, exclamation, null);
+            SyntaxToken exclamationExclamation = null;
+            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+            {
+                exclamationExclamation = this.EatToken();
+                if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                {
+                    exclamationExclamation = MergeTokens(exclamationExclamation, this.EatToken(), SyntaxKind.ExclamationExclamationToken);
+                }
```

Comment:
Error case?

Reply:
Just unconditionally call `exclamationExclamation = MergeTokens(exclamationExclamation, this.EatToken(SyntaxKind.ExclamationToken), SyntaxKind.ExclamationExclamationToken);`, regardless of what the current token kind is. We'll report a good error if it's not.

---
In reply to: [466736323](https://github.com/dotnet/roslyn/pull/46520#discussion_r466736323) [](ancestors = 466736323)